File: AMF4/src/test/js/validation.test.js
const amf = require('amf-client-js');
const path = require('path');

const AMF = amf.AMF;
const ProfileNames = amf.ProfileNames;
const MessageStyles = amf.MessageStyles;

beforeAll(() => {
    amf.plugins.features.AMFValidation.register();
    amf.AMF.init();
});

test('Validate RAML 1.0', () => {
    const parser = new amf.Raml10Parser();

    return parser.parseFileAsync(`file://${path.resolve(__dirname, '../../../resources/examples/banking-api-error.raml')}`)
        .then(model => {
            expect(model).not.toBeNull();
            expect(model).toBeDefined();

            // Run RAML default validations on parsed unit (expects one error -> invalid protocols value)
            return AMF.validate(model, ProfileNames.RAML, MessageStyles.RAML)
                .then(report => {
                    console.log(`report.conforms() == ${report.conforms}`)
                    expect(report.conforms).toBe(false)
                })
        });
})

test('Validate RAML 1.0 with custom validation', () => {
    const parser = new amf.Raml10Parser();

    return parser.parseFileAsync(`file://${path.resolve(__dirname, '../../../resources/examples/banking-api-error.raml')}`)
        .then(model => {
            expect(model).not.toBeNull();
            expect(model).toBeDefined();

            const customValidationProfilePath = `file://${path.resolve(__dirname, '../../../resources/validation_profile.raml')}`;

            // Run RAML custom validations with a validation profile that accepts the previously invalid protocol value
            // TODO: Fix
            // Failed: bc {
            //   "Hg": null,
            //   "SJa": true,
            //   "Xs": "Cannot load a custom validation profile for this",
            //   "ZX": null,
            //   "stackdata": [Circular],
            // }
            return AMF.loadValidationProfile(customValidationProfilePath)
                .then(customProfile => {
                    return AMF.validate(model, customProfile, MessageStyles.RAML)
                        .then(report => {
                            console.log(`report.conforms() == ${report.conforms}`)
                            expect(report.conforms).toBe(true)
                        })
                })
        });
})

File: AMF4/src/test/java/ValidationTest.java
import amf.MessageStyles;
import amf.ProfileName;
import amf.ProfileNames;
import amf.client.AMF;
import amf.client.model.document.BaseUnit;
import amf.client.parse.Raml10Parser;
import amf.client.validate.ValidationReport;
import org.junit.BeforeClass;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.junit.Assert.*;

public class ValidationTest {

    @BeforeClass
    public static void setup() throws ExecutionException, InterruptedException {
        AMF.init().get();
    }


    @Test
    public void validateRaml() throws ExecutionException, InterruptedException {
        final Raml10Parser parser = new Raml10Parser();

        final BaseUnit model = parser.parseFileAsync("file://resources/examples/banking-api-error.raml").get();
        assertNotNull(model);

        // Run RAML default validations on parsed unit (expects one error -> invalid protocols value)
        final ValidationReport report = AMF.validate(model, ProfileNames.RAML(), MessageStyles.RAML()).get();
        System.out.println("report.conforms() == " + report.conforms());
        assertFalse(report.conforms());
    }

    @Test
    public void validateRamlWithCustomValidation() throws ExecutionException, InterruptedException {
        final Raml10Parser parser = new Raml10Parser();

        final BaseUnit model = parser.parseFileAsync("file://resources/examples/banking-api-error.raml").get();
        assertNotNull(model);

        // Run RAML custom validations with a validation profile that accepts the previously invalid protocol value
        final ProfileName customProfile = AMF.loadValidationProfile("file://resources/validation_profile.raml").get();

        final ValidationReport report = AMF.validate(model, customProfile, MessageStyles.RAML()).get();
        System.out.println("report.conforms() == " + report.conforms());
        assertTrue(report.conforms());
    }
}
